confdgnmi/src/confd_gnmi_adapter.py

```diff
@@ -58,126 +58,238 @@ def set(self, prefix, path, val):
         """
         pass
 
-
     class SubscriptionHandler(ABC):
 
-        def __init__(self, adapter):
+        class SubscriptionEvent(Enum):
+            SAMPLE = 0
+            SEND_CHANGES = 1
+            FINISH = 10
+
+        def __init__(self, adapter, subscription_list):
             self.adapter = adapter
+            self.subscription_list = subscription_list
+            self.read_queue: Queue = None
 
-        subscription_list = None
+        @abstractmethod
+        def get_sample(self, path, prefix) -> []:
+            """
+            Create gNMI subscription updates for given path and prefix
+            :param path: gNMI path for updates
+            :param prefix: gNMI prefix
+            #TODO do we need to return array or would just one Update be enough?
+            :return: gNMI update array
+            """
+            pass
 
-        class SubscriptionStreamEventType(Enum):
-            GET = 0
-            FINISH = 1
+        # SubscriptionHandler abstract methods - start
+        @abstractmethod
+        def add_path_for_monitoring(self, path, prefix):
+            """
+            Add this path for monitoring for changes
+            Monitoring must be stopped.
+            :param path:
+            :param prefix:
+            :return:
+            """
+            pass
 
-        subscription_stream_event_type: SubscriptionStreamEventType = None
-        thread_event: Event = None
+        @abstractmethod
+        def get_monitored_changes(self) -> []:
+            """
+            Get gNMI subscription updates for changed values (indicated by
+            self.SubscriptionEvent.PROCESS_CHANGES)
```

Comment:
`PROCESS_CHANGES` - did you mean `SEND_CHANGES`?

Reply:
Yes, I'll remove reference to event (enum)